File: SDKs/amazonaws.com/codecommit/2015-04-13/java/lib/src/main/java/openapisdk/models/shared/ApprovalStateChangedEventMetadata.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ApprovalStateChangedEventMetadata
 * Returns information about a change in the approval state for a pull request.
**/
public class ApprovalStateChangedEventMetadata {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("approvalStatus")
    public ApprovalStateEnum approvalStatus;
    public ApprovalStateChangedEventMetadata withApprovalStatus(ApprovalStateEnum approvalStatus) {
        this.approvalStatus = approvalStatus;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("revisionId")
    public String revisionId;
    public ApprovalStateChangedEventMetadata withRevisionId(String revisionId) {
        this.revisionId = revisionId;
        return this;
    }
}

File: SDKs/amazonaws.com/codecommit/2015-04-13/python/src/sdk/models/shared/approvalstatechangedeventmetadata.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import approvalstate_enum as shared_approvalstate_enum


@dataclass_json
@dataclasses.dataclass
class ApprovalStateChangedEventMetadata:
    r"""ApprovalStateChangedEventMetadata
    Returns information about a change in the approval state for a pull request.
    """
    
    approval_status: Optional[shared_approvalstate_enum.ApprovalStateEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('approvalStatus') }})
    revision_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('revisionId') }})
    
